compiler/resolution/lowerIterators.cpp

```diff
@@ -57,9 +57,11 @@ FnSymbol* getTheIteratorFn(CallExpr* call) {
 FnSymbol* getTheIteratorFn(Type* icType)
 {
   // the asserts document the current state
+  AggregateType* icTypeAgg = toAggregateType(icType);
```

Comment:
I think lines 60 and 61 imply the icType must be an aggregate type.  So one
wonders why it accepts a static Type?